graders/python/python_autograder/pl_result.py

```diff
@@ -95,6 +95,10 @@ def addError(self, test: unittest.TestCase, err: Any) -> None:  # noqa: N802
                 st_code=Feedback.test.student_code_abs_path,
                 ipynb_key=Feedback.test.ipynb_key,
             )
+        elif isinstance(err[1], GradingTestComplete):
+            # A general test failure has occurred
+            # It's assumed that test case feedback has already been given
+            self.results[-1]["points"] = 0
```

Comment:
I'm probably getting too in the weeds here, but...

`finish`/`finish_test` are both meant to be "neutral" things (I thought), but both the existing handler for `GradingComplete` and this new one for `GradingTestComplete` assign zero points. I'm thinking we want to shift that to the exception handler where we add the "The callable supplied in your code failed with an exception..." message? I acknowledge that this would be considered a breaking change, but I checked and I don't think anyone is using `finish()` like this. Specifically, one would have to have set the score to some non-zero value and then called finish with the expectation that it would re-zero the score, which seems weird already, but I really can't see anyone that's doing this.

Reply:
This is exactly why I wanted to change the language of `finish` to be non-neutral :)

Reply:
I think shifting the point assignment would be a good change, as long as it isn't breaking.

Reply:
That seems reasonable to me as well. I'll be busy for the next couple hours so feel free to push the change in the meantime. If not I'll take a look tonight.

Reply:
> This is exactly why I wanted to change the language of `finish` to be non-neutral :)

Fair! I hadn't quite pieced together the points thing when you called that out, apologies 😅

I'll push up the change shortly.

Reply:
I moved around the points assignment in e975af4.